src/locationSource/LocationSource.js

```diff
@@ -0,0 +1,74 @@
+import GpsApiLocationSource from "./GpsApiLocationSource";
+import IpBasedLocationSource from "./IpBasedLocationSource";
+
+const GPS_TIMEOUT = 4000;
```

Comment:
Takie rzeczy powinny się chyba raczej pojawiać w konstruktorze LocationSource. Plus pominąłeś część argumentów IpBasedLocationSource, więc żeby dopisać/zmienić stosowane API sprawdzające IP to trzeba głęboko w metodach Twojej klasy grzebać.

Reply:
Poprawione, dużo łatwiej teraz znaleźć i dodać nowe API do sprawdzania - wystarczy dodać nowy obiekt do statycznego gettera `IP_PROMISES_ARRAY`, który potem jest mapowany na Promise. Dużo bardziej elastyczne mi się to teraz wydaje, dzięki za uwagę.